modules/OCPP/OCPP.cpp

```diff
@@ -98,20 +105,39 @@ void OCPP::set_external_limits(const std::map<int32_t, ocpp::v16::EnhancedChargi
             schedule_import.push_back(schedule_req_entry);
         }
         limits.schedule_import.emplace(schedule_import);
+        auto& evse_sink = this->get_evse_sink_by_evse_id(connector_id);
+        evse_sink.call_set_external_limits(limits);
+    }
+}
 
-        if (connector_id == 0) {
-            if (!this->r_connector_zero_sink.empty()) {
-                EVLOG_debug << "OCPP sets the following external limits for connector 0: \n" << limits;
-                this->r_connector_zero_sink.at(0)->call_set_external_limits(limits);
-            } else {
-                EVLOG_debug << "OCPP cannot set external limits for connector 0. No "
-                               "sink is configured.";
-            }
-        } else {
-            EVLOG_debug << "OCPP sets the following external limits for connector " << connector_id << ": \n" << limits;
-            this->r_evse_manager.at(connector_id - 1)->call_set_external_limits(limits);
+bool OCPP::is_evse_sink_configured(const int32_t evse_id) {
+    for (const auto& evse_sink : this->r_evse_energy_sink) {
+        if (not evse_sink->get_mapping().has_value()) {
+            EVLOG_critical << "Please configure an evse mapping your configuration file for the connected "
+                              "r_evse_energy_sink with module_id: "
+                           << evse_sink->module_id;
+            throw std::runtime_error("No mapping configured for evse_id: " + evse_id);
+        }
+        if (evse_sink->get_mapping().value().evse == evse_id) {
+            return true;
         }
     }
+    return false;
+}
+
+external_energy_limitsIntf& OCPP::get_evse_sink_by_evse_id(const int32_t evse_id) {
+    for (const auto& evse_sink : this->r_evse_energy_sink) {
+        if (not evse_sink->get_mapping().has_value()) {
+            EVLOG_critical << "Please configure an evse mapping your configuration file for the connected "
+                              "r_evse_energy_sink with module_id: "
+                           << evse_sink->module_id;
+            throw std::runtime_error("No mapping configured for evse_id: " + evse_id);
+        }
+        if (evse_sink->get_mapping().value().evse == evse_id) {
+            return *evse_sink;
+        }
+    }
+    throw std::runtime_error("No mapping configured for evse");
```

Comment:
This looks like duplicated code.  It could be factored out into a separate function and reused by the OCPP and API module